x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/assembler/BoxedQueryRequest.java

```diff
@@ -70,6 +90,78 @@ public BoxedQueryRequest to(Ordinal end) {
         return this;
     }
 
+    /**
+     * Sets keys / terms to filter on.
+     * Accepts the unwrapped SequenceKey as a list of values matching an instance of a given
+     * event.
+     * Can be removed through null.
+     */
+    public BoxedQueryRequest keys(List<List<Object>> values) {
+        List<QueryBuilder> newFilters;
+
+        if (values == null || values.isEmpty()) {
+            // no keys have been specified and none have been set
+            if (keyFilters == null || keyFilters.isEmpty()) {
+                return this;
+            }
+            newFilters = emptyList();
+        }
+        else {
+            // iterate on all possible values for a given key
+            newFilters = new ArrayList<>(values.size());
+            for (int keyIndex = 0; keyIndex < keys.size(); keyIndex++) {
+                String key = keys.get(keyIndex);
```

Comment:
The `key` here is needed further down after the max_terms size check is performed on `keyValues` size. You could move the `key` initialization further down after you initialize the `query` QueryBuilder with `null`.